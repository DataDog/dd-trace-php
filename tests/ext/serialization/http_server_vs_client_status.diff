     === SERVER SPAN WITH 404 ===
002- === CLIENT SPAN WITH 404 ===
003- === RESULTS ===
004- string(17) "Client 404 error:"
005- int(1)
006- string(17) "Server 404 error:"
007- NULL
002+ 
003+ Fatal error: Uncaught Error: Call to undefined function DDTrace\start_span() in /home/circleci/app/tests/ext/serialization/http_server_vs_client_status.php:4
004+ Stack trace:
005+ #0 {main}
006+   thrown in /home/circleci/app/tests/ext/serialization/http_server_vs_client_status.php on line 4
