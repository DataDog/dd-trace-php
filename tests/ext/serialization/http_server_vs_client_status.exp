=== SERVER SPAN WITH 404 ===
=== CLIENT SPAN WITH 404 ===
=== RESULTS ===
string(17) "Client 404 error:"
int(1)
string(17) "Server 404 error:"
NULL