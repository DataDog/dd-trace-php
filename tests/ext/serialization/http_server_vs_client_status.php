<?php
// 1. Server span with 404 - should NOT be error
echo "=== SERVER SPAN WITH 404 ===\n";
$span_server = \DDTrace\start_span();
$span_server->meta['http.status_code'] = 404;
$span_server->meta['debug_id'] = 'SERVER_SPAN';
\DDTrace\close_span();

// 2. Client span with 404 - should be error
echo "=== CLIENT SPAN WITH 404 ===\n";
$span_client = \DDTrace\start_span();
$span_client->meta['http.status_code'] = 404;
$span_client->meta['span.kind'] = 'client';
$span_client->meta['debug_id'] = 'CLIENT_SPAN';
\DDTrace\close_span();

// Get serialized spans
$serialized = dd_trace_serialize_closed_spans();

// Check results
echo "=== RESULTS ===\n";
// Find spans by debug ID
$client_span = null;
$server_span = null;
foreach ($serialized as $span) {
    if (isset($span['meta']['debug_id'])) {
        if ($span['meta']['debug_id'] === 'CLIENT_SPAN') {
            $client_span = $span;
        } elseif ($span['meta']['debug_id'] === 'SERVER_SPAN') {
            $server_span = $span;
        }
    }
}

var_dump("Client 404 error:", isset($client_span['error']) ? $client_span['error'] : null);
var_dump("Server 404 error:", isset($server_span['error']) ? $server_span['error'] : null);
?>
