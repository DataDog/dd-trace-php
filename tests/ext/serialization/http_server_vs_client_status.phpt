--TEST--
Client and server spans use their respective configurations
--ENV--
DD_TRACE_HTTP_SERVER_ERROR_STATUSES=500-599
DD_TRACE_HTTP_CLIENT_ERROR_STATUSES=400-499
--FILE--
<?php
// 1. Server span with 404 - should NOT be error
echo "=== SERVER SPAN WITH 404 ===\n";
$span_server = \DDTrace\start_span();
$span_server->meta['http.status_code'] = 404;
$span_server->meta['debug_id'] = 'SERVER_SPAN';
\DDTrace\close_span();

// 2. Client span with 404 - should be error
echo "=== CLIENT SPAN WITH 404 ===\n";
$span_client = \DDTrace\start_span();
$span_client->meta['http.status_code'] = 404;
$span_client->meta['span.kind'] = 'client';
$span_client->meta['debug_id'] = 'CLIENT_SPAN';
\DDTrace\close_span();

// Get serialized spans
$serialized = dd_trace_serialize_closed_spans();

// Check results
echo "=== RESULTS ===\n";
// Find spans by debug ID
$client_span = null;
$server_span = null;
foreach ($serialized as $span) {
    if (isset($span['meta']['debug_id'])) {
        if ($span['meta']['debug_id'] === 'CLIENT_SPAN') {
            $client_span = $span;
        } elseif ($span['meta']['debug_id'] === 'SERVER_SPAN') {
            $server_span = $span;
        }
    }
}

var_dump("Client 404 error:", isset($client_span['error']) ? $client_span['error'] : null);
var_dump("Server 404 error:", isset($server_span['error']) ? $server_span['error'] : null);
?>
--EXPECTF--
=== SERVER SPAN WITH 404 ===
=== CLIENT SPAN WITH 404 ===
=== RESULTS ===
string(17) "Client 404 error:"
int(1)
string(17) "Server 404 error:"
NULL