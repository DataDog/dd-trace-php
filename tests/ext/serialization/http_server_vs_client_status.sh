#!/bin/sh

export RUNTIME_DEPS='apache2     apache2-dev     ca-certificates     clang-format     curl     debian-goodies     git     less     lsof     netbase     netcat     nginx     strace     sudo     unzip     vim     xz-utils     zip'
export HOSTNAME='9ad8a3847fc6'
export PHP_VERSION='8.3'
export APACHE_CONFDIR='/etc/apache2'
export DATADOG_HAVE_DEV_ENV='1'
export HTTPBIN_HOSTNAME='httpbin_integration'
export PWD='/home/circleci/app'
export PHP_SRC_DIR='/usr/local/src/php'
export RUSTC_BOOTSTRAP='1'
export DD_TRACE_DOCKER_DEBUG='1'
export HOME='/home/circleci'
export LANG='C.UTF-8'
export PHP_IDE_CONFIG='serverName=docker'
export LS_COLORS='rs=0:di=01;34:ln=01;36:mh=00:pi=40;33:so=01;35:do=01;35:bd=40;33;01:cd=40;33;01:or=40;31;01:mi=00:su=37;41:sg=30;43:ca=30;41:tw=30;42:ow=34;42:st=37;44:ex=01;32:*.tar=01;31:*.tgz=01;31:*.arc=01;31:*.arj=01;31:*.taz=01;31:*.lha=01;31:*.lz4=01;31:*.lzh=01;31:*.lzma=01;31:*.tlz=01;31:*.txz=01;31:*.tzo=01;31:*.t7z=01;31:*.zip=01;31:*.z=01;31:*.dz=01;31:*.gz=01;31:*.lrz=01;31:*.lz=01;31:*.lzo=01;31:*.xz=01;31:*.zst=01;31:*.tzst=01;31:*.bz2=01;31:*.bz=01;31:*.tbz=01;31:*.tbz2=01;31:*.tz=01;31:*.deb=01;31:*.rpm=01;31:*.jar=01;31:*.war=01;31:*.ear=01;31:*.sar=01;31:*.rar=01;31:*.alz=01;31:*.ace=01;31:*.zoo=01;31:*.cpio=01;31:*.7z=01;31:*.rz=01;31:*.cab=01;31:*.wim=01;31:*.swm=01;31:*.dwm=01;31:*.esd=01;31:*.jpg=01;35:*.jpeg=01;35:*.mjpg=01;35:*.mjpeg=01;35:*.gif=01;35:*.bmp=01;35:*.pbm=01;35:*.pgm=01;35:*.ppm=01;35:*.tga=01;35:*.xbm=01;35:*.xpm=01;35:*.tif=01;35:*.tiff=01;35:*.png=01;35:*.svg=01;35:*.svgz=01;35:*.mng=01;35:*.pcx=01;35:*.mov=01;35:*.mpg=01;35:*.mpeg=01;35:*.m2v=01;35:*.mkv=01;35:*.webm=01;35:*.ogm=01;35:*.mp4=01;35:*.m4v=01;35:*.mp4v=01;35:*.vob=01;35:*.qt=01;35:*.nuv=01;35:*.wmv=01;35:*.asf=01;35:*.rm=01;35:*.rmvb=01;35:*.flc=01;35:*.avi=01;35:*.fli=01;35:*.flv=01;35:*.gl=01;35:*.dl=01;35:*.xcf=01;35:*.xwd=01;35:*.yuv=01;35:*.cgm=01;35:*.emf=01;35:*.ogv=01;35:*.ogx=01;35:*.aac=00;36:*.au=00;36:*.flac=00;36:*.m4a=00;36:*.mid=00;36:*.midi=00;36:*.mka=00;36:*.mp3=00;36:*.mpc=00;36:*.ogg=00;36:*.ra=00;36:*.wav=00;36:*.oga=00;36:*.opus=00;36:*.spx=00;36:*.xspf=00;36:'
export CARGO_HOME='/rust/cargo'
export CMAKE_VERSION='3.24.4'
export PHP_INSTALL_DIR='/opt/php'
export GOROOT='/go'
export PHPIZE_DEPS='autoconf     bison     clang-17     cmake     dpkg-dev     file     gnupg     gpg     file     libc-dev     make     pkg-config     re2c'
export REDIS_HOSTNAME='redis_integration'
export LSAN_OPTIONS='suppressions=/home/circleci/suppr.txt,print_suppressions=0,use_tls=0'
export TERM='xterm'
export ACCEPT_EULA='Y'
export RUSTUP_HOME='/rust/rustup'
export SHLVL='1'
export COMPOSER_MEMORY_LIMIT='-1'
export APACHE_ENVVARS='/etc/apache2/envvars'
export DD_SPAWN_WORKER_USE_EXEC='1'
export PATH='/go/bin:/rust/cargo/bin:/usr/local/sbin:/usr/local/bin:/usr/sbin:/usr/bin:/sbin:/bin'
export DEVLIBS='libclang-17-dev     libclang-rt-17-dev     llvm-17-dev     lld-17     libbrotli-dev     libcurl4-openssl-dev     libedit-dev     libffi-dev     libfreetype6-dev     libicu-dev     libjpeg-dev     libmcrypt-dev     libmemcached-dev     libodbc1     libonig-dev     libpq-dev     libpng-dev     librabbitmq-dev     librdkafka-dev     libsodium-dev     libsqlite3-dev     libssl-dev     libwebp-dev     libxml2-dev     libxslt1-dev     libzip-dev     libc6-dbg     odbcinst1debian2     zlib1g-dev     unixodbc-dev     unixodbc'
export DDAGENT_HOSTNAME='ddagent_integration'
export DEBIAN_FRONTEND='noninteractive'
export OLDPWD='/home/circleci/app/target/debug/build'
export _='/usr/local/bin/php'
export TEMP='/tmp'
export TEST_PHP_EXECUTABLE='/opt/php/debug/bin/php'
export TEST_PHP_EXECUTABLE_ESCAPED=''\''/opt/php/debug/bin/php'\'''
export TEST_PHP_CGI_EXECUTABLE='/opt/php/debug/bin/php-cgi'
export TEST_PHP_CGI_EXECUTABLE_ESCAPED=''\''/opt/php/debug/bin/php-cgi'\'''
export TEST_PHPDBG_EXECUTABLE=''
export TEST_PHPDBG_EXECUTABLE_ESCAPED=''\'''\'''
export REDIRECT_STATUS='1'
export QUERY_STRING=''
export PATH_TRANSLATED='/home/circleci/app/tests/ext/serialization/http_server_vs_client_status.php'
export SCRIPT_FILENAME='/home/circleci/app/tests/ext/serialization/http_server_vs_client_status.php'
export REQUEST_METHOD='GET'
export CONTENT_TYPE=''
export CONTENT_LENGTH=''
export TZ=''
export DD_TRACE_HTTP_SERVER_ERROR_STATUSES='500-599'
export DD_TRACE_HTTP_CLIENT_ERROR_STATUSES='400-499'
export TEST_PHP_EXTRA_ARGS='  -d "output_handler=" -d "open_basedir=" -d "disable_functions=" -d "output_buffering=Off" -d "error_reporting=32767" -d "display_errors=1" -d "display_startup_errors=1" -d "log_errors=0" -d "html_errors=0" -d "track_errors=0" -d "report_memleaks=1" -d "report_zend_debug=0" -d "docref_root=" -d "docref_ext=.html" -d "error_prepend_string=" -d "error_append_string=" -d "auto_prepend_file=" -d "auto_append_file=" -d "ignore_repeated_errors=0" -d "precision=14" -d "serialize_precision=-1" -d "memory_limit=128M" -d "opcache.fast_shutdown=0" -d "opcache.file_update_protection=0" -d "opcache.revalidate_freq=0" -d "opcache.jit_hot_loop=1" -d "opcache.jit_hot_func=1" -d "opcache.jit_hot_return=1" -d "opcache.jit_hot_side_exit=1" -d "zend.assertions=1" -d "zend.exception_ignore_args=0" -d "zend.exception_string_param_max_len=15" -d "short_open_tag=0"'
export HTTP_COOKIE=''

case "$1" in
"gdb")
    gdb --args '/opt/php/debug/bin/php'    -d "output_handler=" -d "open_basedir=" -d "disable_functions=" -d "output_buffering=Off" -d "error_reporting=32767" -d "display_errors=1" -d "display_startup_errors=1" -d "log_errors=0" -d "html_errors=0" -d "track_errors=0" -d "report_memleaks=1" -d "report_zend_debug=0" -d "docref_root=" -d "docref_ext=.html" -d "error_prepend_string=" -d "error_append_string=" -d "auto_prepend_file=" -d "auto_append_file=" -d "ignore_repeated_errors=0" -d "precision=14" -d "serialize_precision=-1" -d "memory_limit=128M" -d "opcache.fast_shutdown=0" -d "opcache.file_update_protection=0" -d "opcache.revalidate_freq=0" -d "opcache.jit_hot_loop=1" -d "opcache.jit_hot_func=1" -d "opcache.jit_hot_return=1" -d "opcache.jit_hot_side_exit=1" -d "zend.assertions=1" -d "zend.exception_ignore_args=0" -d "zend.exception_string_param_max_len=15" -d "short_open_tag=0" -f "/home/circleci/app/tests/ext/serialization/http_server_vs_client_status.php"  2>&1
    ;;
"lldb")
    lldb -- '/opt/php/debug/bin/php'    -d "output_handler=" -d "open_basedir=" -d "disable_functions=" -d "output_buffering=Off" -d "error_reporting=32767" -d "display_errors=1" -d "display_startup_errors=1" -d "log_errors=0" -d "html_errors=0" -d "track_errors=0" -d "report_memleaks=1" -d "report_zend_debug=0" -d "docref_root=" -d "docref_ext=.html" -d "error_prepend_string=" -d "error_append_string=" -d "auto_prepend_file=" -d "auto_append_file=" -d "ignore_repeated_errors=0" -d "precision=14" -d "serialize_precision=-1" -d "memory_limit=128M" -d "opcache.fast_shutdown=0" -d "opcache.file_update_protection=0" -d "opcache.revalidate_freq=0" -d "opcache.jit_hot_loop=1" -d "opcache.jit_hot_func=1" -d "opcache.jit_hot_return=1" -d "opcache.jit_hot_side_exit=1" -d "zend.assertions=1" -d "zend.exception_ignore_args=0" -d "zend.exception_string_param_max_len=15" -d "short_open_tag=0" -f "/home/circleci/app/tests/ext/serialization/http_server_vs_client_status.php"  2>&1
    ;;
"valgrind")
    USE_ZEND_ALLOC=0 valgrind $2 '/opt/php/debug/bin/php'    -d "output_handler=" -d "open_basedir=" -d "disable_functions=" -d "output_buffering=Off" -d "error_reporting=32767" -d "display_errors=1" -d "display_startup_errors=1" -d "log_errors=0" -d "html_errors=0" -d "track_errors=0" -d "report_memleaks=1" -d "report_zend_debug=0" -d "docref_root=" -d "docref_ext=.html" -d "error_prepend_string=" -d "error_append_string=" -d "auto_prepend_file=" -d "auto_append_file=" -d "ignore_repeated_errors=0" -d "precision=14" -d "serialize_precision=-1" -d "memory_limit=128M" -d "opcache.fast_shutdown=0" -d "opcache.file_update_protection=0" -d "opcache.revalidate_freq=0" -d "opcache.jit_hot_loop=1" -d "opcache.jit_hot_func=1" -d "opcache.jit_hot_return=1" -d "opcache.jit_hot_side_exit=1" -d "zend.assertions=1" -d "zend.exception_ignore_args=0" -d "zend.exception_string_param_max_len=15" -d "short_open_tag=0" -f "/home/circleci/app/tests/ext/serialization/http_server_vs_client_status.php"  2>&1
    ;;
"rr")
    rr record $2 '/opt/php/debug/bin/php'    -d "output_handler=" -d "open_basedir=" -d "disable_functions=" -d "output_buffering=Off" -d "error_reporting=32767" -d "display_errors=1" -d "display_startup_errors=1" -d "log_errors=0" -d "html_errors=0" -d "track_errors=0" -d "report_memleaks=1" -d "report_zend_debug=0" -d "docref_root=" -d "docref_ext=.html" -d "error_prepend_string=" -d "error_append_string=" -d "auto_prepend_file=" -d "auto_append_file=" -d "ignore_repeated_errors=0" -d "precision=14" -d "serialize_precision=-1" -d "memory_limit=128M" -d "opcache.fast_shutdown=0" -d "opcache.file_update_protection=0" -d "opcache.revalidate_freq=0" -d "opcache.jit_hot_loop=1" -d "opcache.jit_hot_func=1" -d "opcache.jit_hot_return=1" -d "opcache.jit_hot_side_exit=1" -d "zend.assertions=1" -d "zend.exception_ignore_args=0" -d "zend.exception_string_param_max_len=15" -d "short_open_tag=0" -f "/home/circleci/app/tests/ext/serialization/http_server_vs_client_status.php"  2>&1
    ;;
*)
    '/opt/php/debug/bin/php'    -d "output_handler=" -d "open_basedir=" -d "disable_functions=" -d "output_buffering=Off" -d "error_reporting=32767" -d "display_errors=1" -d "display_startup_errors=1" -d "log_errors=0" -d "html_errors=0" -d "track_errors=0" -d "report_memleaks=1" -d "report_zend_debug=0" -d "docref_root=" -d "docref_ext=.html" -d "error_prepend_string=" -d "error_append_string=" -d "auto_prepend_file=" -d "auto_append_file=" -d "ignore_repeated_errors=0" -d "precision=14" -d "serialize_precision=-1" -d "memory_limit=128M" -d "opcache.fast_shutdown=0" -d "opcache.file_update_protection=0" -d "opcache.revalidate_freq=0" -d "opcache.jit_hot_loop=1" -d "opcache.jit_hot_func=1" -d "opcache.jit_hot_return=1" -d "opcache.jit_hot_side_exit=1" -d "zend.assertions=1" -d "zend.exception_ignore_args=0" -d "zend.exception_string_param_max_len=15" -d "short_open_tag=0" -f "/home/circleci/app/tests/ext/serialization/http_server_vs_client_status.php"  2>&1
    ;;
esac