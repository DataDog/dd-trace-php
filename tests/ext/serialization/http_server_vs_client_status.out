=== SERVER SPAN WITH 404 ===

Fatal error: Uncaught Error: Call to undefined function DDTrace\start_span() in /home/circleci/app/tests/ext/serialization/http_server_vs_client_status.php:4
Stack trace:
#0 {main}
  thrown in /home/circleci/app/tests/ext/serialization/http_server_vs_client_status.php on line 4